lib/actions/field-trip.js

```diff
@@ -249,17 +406,355 @@ function planInbound (request) {
+          // not used by another vehicle, so not used by this other field trip
+          if (!sameVehicleTrip) return
+
+          // check if the stop ranges overlap. It is OK if one field trip begins
+          // where the other ends.
+          if (
+            leg.from.stopIndex >= tripInUse.toStopIndex ||
+              leg.to.stopIndex <= tripInUse.fromStopIndex
+          ) {
+            // legs do not overlap, so not used by this other field trip
+            return
+          }
+
+          // ranges overlap! Add number of passengers on this other field trip
+          // to total capacity in use
+          capacityInUse += tripInUse.passengers
+        })
+
+        // check if the remaining capacity on this trip is enough to allow more
+        // field trip passengers on board
+        const legModeCapacity = getFieldTripGroupCapacityForMode(
+          fieldTripModuleConfig,
+          leg.mode
+        )
+        let remainingTripCapacity = legModeCapacity - capacityInUse
+        if (remainingTripCapacity < minimumAllowableRemainingCapacity) {
+          // This trip is already too "full" to allow any addition field trips
+          // on board. Ban this trip in future searches and don't use this
+          // itinerary in final results (set trip and itinerary capacity to 0).
+          remainingTripCapacity = 0
+        }
+
+        // always ban trips found in itineraries so that subsequent searches
+        // don't encounter them.
+        // TODO: a more advanced way of doing things might be to ban trip
+        //        sequences to not find the same exact sequence, but also
+        //        individual trips that are too full.
+        tripsToBanInSubsequentSearches.push(tripId)
+
+        itineraryCapacity = Math.min(itineraryCapacity, remainingTripCapacity)
+      })
+
+      if (itineraryCapacity > 0) {
+        // itinerary is possible, add to list and update remaining group size.
+        // A field trip response is guaranteed to have only one itinerary, so it
+        // ok to set the itinerary by response as an array with a single
+        // itinerary.
+        assignedItinerariesByResponse[responseIdx] = [{
+          ...itinerary,
+          fieldTripGroupSize: Math.min(itineraryCapacity, remainingGroupSize)
+        }]
+        remainingGroupSize -= itineraryCapacity
+      }
+    })
+  })
+
+  return {
+    assignedItinerariesByResponse,
+    remainingGroupSize,
+    tripsToBanInSubsequentSearches
   }
 }
 
+/**
+ * Formats a field trip's travel date into the given format.
+ */
+function getFormattedRequestTravelDate (
+  request,
+  dateFormat = OTP_API_DATE_FORMAT
+) {
+  return moment(request.travelDate).format(dateFormat)
+}
+
+// These can be overridden in the field trip module config.
+const defaultFieldTripModeCapacities = {
+  'TRAM': 80,
+  'SUBWAY': 120,
+  'RAIL': 80,
+  'BUS': 40,
+  'FERRY': 100,
+  'CABLE_CAR': 20,
+  'GONDOLA': 15,
+  'FUNICULAR': 20
+}
+const unknownModeCapacity = 15
```

Comment:
Should this constant be spelled uppercase?

Reply:
[According to Airbnb, no](https://github.com/airbnb/javascript/#naming--uppercase)?